examples/xtt_client.c

```diff
@@ -405,7 +407,7 @@ int initialize_server_id(xtt_identity_type *intended_server_id,
     int read_ret;
 
     // Set server's id from file/NVRAM
-    if (use_tpm) {
+    if (use_tpm && tcti_context) {
```

Comment:
What's the reason for including the non-NULL check on tcti_context in each of these?

I'm not saying it's bad to check for non-NULL. But we *should* never have `use_tpm == true` but not have initialized the context. If you were running into that, that's a problem.

Reply:
The compiler was complaining about the `tcti_context` variable not being used. I saw two options: either ifdef out the parameter in each function, which was kind of messy, or check that it was non-NULL.

Reply:
Ah, I see.

For future reference, an idiom you can use to shut-up a compiler warning about unused parameters (if you know you don't need to use them, or during testing) is to use
```
(void) un_used_parameter;
```
In other words, you're casting the parameter to a void.

For this specific situation, I think we need to re-organize the client code some, to not need this. However, I'll open a separate issue about that. This looks good to me.